packages/vscode-extension/src/devices/preview.ts

```diff
@@ -60,33 +107,43 @@ export class Preview implements Disposable {
             this.streamURL = match[1];
             resolve(this.streamURL);
           }
-        } else if (line.includes("video_ready replay") || line.includes("video_error replay")) {
-          // video response format for replays looks as follows:
-          // video_ready replay <HTTP_URL> <FILE_URL>
-          // video_error replay <Error message>
-          const videoReadyMatch = line.match(/video_ready replay (\S+) (\S+)/);
-          const videoErrorMatch = line.match(/video_error replay (.*)/);
-
-          const handlers = this.lastReplayPromise;
-          this.lastReplayPromise = undefined;
+        } else if (line.includes("video_ready") || line.includes("video_error")) {
+          // video response format for recordings looks as follows:
+          // video_ready <VIDEO_ID> <HTTP_URL> <FILE_URL>
+          // video_error <VIDEO_ID> <Error message>
+          const videoReadyMatch = line.match(/video_ready (\S+) (\S+) (\S+)/);
+          const videoErrorMatch = line.match(/video_error (\S+) (.*)/);
+
+          const videoId = videoReadyMatch
+            ? videoReadyMatch[1]
+            : videoErrorMatch
+            ? videoErrorMatch[1]
+            : "";
+
+          if (!this.videoRecordingPromises.has(videoId as VideoRecordingPromiseType)) {
+            throw new Error(`Invalid video ID: ${videoId}`);
+          }
 
+          const handlers = this.videoRecordingPromises.get(videoId as VideoRecordingPromiseType);
```

Comment:
same comment about `as`. We should just get it earlier and throw when `handlers` is undefined rather than using `.has`